stm32-modules/include/flex-stacker/flex-stacker/tof_utils.hpp

```diff
@@ -0,0 +1,47 @@
+#include <array>
+#include <cstdint>
+#include <string>
+
+constexpr uint8_t HISTOGRAM_LEN = 135;
+// start at -6 so that the first byte will add 8 bits to it.
+constexpr int PROCESSED_BITS = -6;
+// Bits in a byte
+constexpr int ONE_BYTE = 8;
+// base64 encoding hhas to be a multiple of 4.
+constexpr int PADDING_MULTIPLE = 4;
+// mask to produce 1 base64 char
+constexpr int BASE64_CHAR_MASK = 0x3F;
+
+// Base64 encoding characters
+static const std::string BASE64_ALPHABET =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+inline auto base64_encode(const std::array<uint8_t, HISTOGRAM_LEN>& data)
+    -> std::string {
+    std::string encoded;
```

Comment:
this kind of use of a `std::string` will heap-allocate its internal storage, possibly multiple times depending on the implementation since we're repeatedly calling `push_back` on it. We should use a cstring or something with a fixed initial size and store the size somewhere or null-terminate it so we don't allocate.